libs/events/src/types/events.ts

```diff
@@ -1,13 +1,18 @@
 import { OnExecutedOrderPayload, OnPostedOrderPayload, OnRejectedOrderPayload } from './orders'
 import { OnToastMessagePayload } from './toastMessages'
 
-export type EventNames = 'updateWidgetConfig' | 'onToastMessage' | 'onPostedOrder' | 'onExecutedOrder'
+// export type CowEvents = 'updateWidgetConfig' | 'onToastMessage' | 'onPostedOrder' | 'onExecutedOrder'
+export enum CowEvents {
+  ON_TOAST_MESSAGE = 'ON_TOAST_MESSAGE',
+  ON_POSTED_ORDER = 'ON_POSTED_ORDER',
+  ON_EXECUTED_ORDER = 'ON_EXECUTED_ORDER',
+  ON_REJECTED_ORDER = 'ON_REJECTED_ORDER',
+}
 
 // Define types for event payloads
 export interface EventPayloads {
```

Comment:
What about a Record<CowEvents, PayloadBaseType> to make sure no enum property is forgotten?

Reply:
Very good point! I will do so

Reply:
mm, I don't know how to do this one. `Record<CowEvents, PayloadBaseType>` don't work as each payload is independent, you need to define it on a per-case basis. 

Shouldn't be too bad though. When you add an event, you will notice you need to add it, and if you forget, you get already now a typescript error:

I've tried:
<img width="888" alt="image" src="https://github.com/cowprotocol/cowswap/assets/2352112/ae734500-4a61-40e8-aca4-8a2cd8963a76">

anyways, if you have an idea how to improve, let me know

Reply:
I don't, but maybe ChatGPT does =P
Or even better, @shoom3301 might!

Reply:
I actually don't know, but I asked one nerd guy about it and he gave me this:
https://gist.github.com/lebedev/60cd207a5985df62c943883aed27e499

I'm not sure does it worth it or not :)

Reply:
Ok, I'm fine with that then

Reply:
Ohh, nice, thanks both!!!

Reply:
980bdf7a5

Reply:
980bdf7a5